pkg/dogstatsd/server.go

```diff
@@ -299,6 +319,16 @@ func (s *Server) parsePacket(packet *listeners.Packet, metricSamples []*metrics.
 				dogstatsdMetricParseErrors.Add(1)
 				continue
 			}
+
+			if s.mapper != nil && len(sample.Tags) == 0 {
```

Comment:
Not sure how problematic this is but this feature won't work well with containers. When the agent can identify the origin container it will [enrich the metric with tags automatically](https://github.com/DataDog/datadog-agent/blob/e42745bf55cf7f628b9e826b0528deef444b5530/pkg/dogstatsd/parser.go#L89-L95), breaking the assumption that no tags mean pure statsd metrics. Even if we removed the check we would not be able to cache the metrics by name since tags could change at any point when a container gets churned.

This identification happens in two cases:
1. When the metrics are sent over UDS
2. When the metrics are sent over UDP and contain a specific tag identifying the container

2\. is not possible in our case since statsd traffic can not contain tags. 1\. is however possible. While the traffic might come out of `airflow` & friends through UDP, it is common to [redirect the UDP traffic to UDS using socat](https://docs.datadoghq.com/developers/dogstatsd/unix_socket/?tab=docker#using-socat-as-a-proxy). Customers do this both to get better tagging but also because it's challenging to set up a UDP pipe between the agent and the application emitting statsd.

If it's doable it would be interesting to check if customers using airflow deploy it in a containerized version.

Reply:
As discussed, we might need to wait this PR https://github.com/DataDog/datadog-agent/pull/4427 that separate the pure parsing and the "tag enrichment" step.

Reply:
PR update to use the mapper before the enrichment.